tests/gnrc_dhcpv6_client/main.c

```diff
@@ -42,6 +42,7 @@ void *_dhcpv6_client_thread(void *args)
     /* configure client to request prefix delegation of /64 subnet
      * interface netif */
     dhcpv6_client_req_ia_pd(netif->pid, 64U);
+    dhcpv6_client_req_ia_na(netif->pid);
```

Comment:
So, assuming that `GNRC_NETIF_AAC_DHCP` was set beforehand somewhere else (e.g. in interface initialization), this would be

```suggestion
    if (netif->ipv6.aac_mode & GNRC_NETIF_AAC_DHCP) {
        dhcpv6_client_req_ia_na(netif->pid);
    }
```

Alternatively, we don't have `dhcpv6_client_req_ia_na` at all and just iterate over all interfaces in `dhcpv6_client_init()` and check if they have `GNRC_NETIF_AAC_DHCP` set to activate IA_NA on those interfaces.

Reply:
> Alternatively, we don't have dhcpv6_client_req_ia_na at all and just iterate over all interfaces in dhcpv6_client_init() and check if they have GNRC_NETIF_AAC_DHCP set to activate IA_NA on those interfaces.

That sounds good! I guess this would also simplify `dhcpv6_client_auto_init`. However, wouldn't this always require a GNRC net interface to use `dhcpv6_client_req_ia_na`?

Reply:
No, if `dhcpv6_client_req_ia_na` would not exist, we would not have to use it in GNRC netif, right? ;-) Or do you mean in case the DHCP flag is set on runtime? In that case, yes there would be a way required to activate IA_NA post-init.